xscen/aggregate.py

```diff
@@ -77,41 +81,142 @@ def climatological_mean(
+    min_periods: Optional[Union[int, float]] = None,
+    stride: int = 1,
+    periods: Optional[Union[list[str], list[list[str]]]] = None,
+    rename_variables: bool = True,
+    to_level: str = "climatology",
+    periods_as_dim: bool = False,
+) -> xr.Dataset:
+    """Perform an operation 'op' over time, for given time periods, respecting the temporal resolution of ds.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Dataset to use for the computation.
+    op : str or dict
+        Operation to perform over time.
+        The operation can be any method name of xarray.core.rolling.DatasetRolling, 'linregress', or a dictionary.
+        If 'op' is a dictionary, the key is the operation name and the value is a dict of kwargs
+        accepted by the operation. While other operations are technically possible,
+        the following are recommended and tested:
+        ['max', 'mean', 'median', 'min', 'std', 'sum', 'var', 'linregress'].
+        Operations beyond methods of xarray.core.rolling.DatasetRolling include:
+
+            - 'linregress' : Computes the linear regression over time, using
+              scipy.stats.linregress and employing years as regressors.
+              The output will have a new dimension 'linreg_param' with coordinates:
+              ['slope', 'intercept', 'rvalue', 'pvalue', 'stderr', 'intercept_stderr'].
+
+        Only one operation per call is supported, so len(op)==1 if a dict.
+    window : int, optional
+        Number of years to use for the rolling operation.
+        If left at None and periods is given, window will be the size of the first period. Hence, if periods are of
+        different lengths, the shortest period should be passed first.
+        If left at None and periods is not given, the window will be the size of the input dataset.
+    min_periods : int or float, optional
+        For the rolling operation, minimum number of years required for a value to be computed.
+        If left at None and the xrfreq is either QS or AS and doesn't start in January,
+        min_periods will be one less than window.
+        Otherwise, if left at None, it will be deemed the same as 'window'.
+        If passed as a float value between 0 and 1, this will be interpreted as the floor of the fraction of the window size.
+    stride : int
+        Stride (in years) at which to provide an output from the rolling window operation.
+    periods : list of str or list of lists of str, optional
+        Either [start, end] or list of [start, end] of continuous periods to be considered.
+        This is needed when the time axis of ds contains some jumps in time.
+        If None, the dataset will be considered continuous.
+    rename_variables : bool
+        If True, '_clim_{op}' will be added to variable names.
+    to_level : str, optional
+        The processing level to assign to the output.
+        If None, the processing level of the inputs is preserved.
+    periods_as_dim : bool
```

Comment:
check out `xs.produce_horizon`.maybe this already does what you want here ?

Reply:
I think that the idea here is that eventually we could fully reverse things, where the `periods_as_dim` functionality is given to `climatological_op`, and `produce_horizon` would simply use `periods_as_dim=True`.

However, it's true that this should be renamed `horizon_as_dim` and use the `horizon` dimension, not `period`. I've talked to @vindelico about this, but forgot to put it here in the PR.

Reply:
@vindelico  In this PR, I would change the name to `horizon` and remove `year`. But the complete integration of `produce_horizon` and `clim_op` can wait for another PR if you want to merge this. Just make an issue to document that we want to do this.

Reply:
Issue opened.